internal/gcsx/storage_handle.go

```diff
@@ -0,0 +1,68 @@
+package gcsx
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"golang.org/x/net/context"
+	"golang.org/x/oauth2"
+	"google.golang.org/api/option"
+)
+
+type storageClient struct {
+	client *storage.Client
+}
+
+type storageClientConfig struct {
+	disableHTTP2        bool
+	maxConnsPerHost     int
+	maxIdleConnsPerHost int
+	tokenSrc            oauth2.TokenSource
+}
+
+// GetStorageClientHandle returns the handle of Go storage client containing
+// customized http client. We can configure the http client using the
+// storageClientConfig parameter.
+func GetStorageClientHandle(ctx context.Context,
+	scConfig storageClientConfig) (sh *storageClient, err error) {
```

Comment:
Please dont use short names for variables like this. Rename it to storageclientConfig.